bitcoin/src/network/constants.rs

```diff
@@ -141,6 +143,34 @@ impl Network {
         };
         Ok(network)
     }
+
+    /// Return the network's chain hash (genesis block hash).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use bitcoin::network::constants::Network;
+    /// use bitcoin::blockdata::constants::ChainHash;
+    ///
+    /// let network = Network::Bitcoin;
+    /// assert_eq!(network.chain_hash(), ChainHash::BITCOIN);
+    /// ```
+    pub fn chain_hash(self) -> ChainHash {
+        ChainHash::using_genesis_block(self)
+    }
+
+    /// Creates a `Network` from the chain hash (genesis block hash).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use bitcoin::network::constants::Network;
+    /// use bitcoin::blockdata::constants::ChainHash;
```

Comment:
As above, FTR we are trying to separate the API from the directory structure by using well place re-exports, its still work in progress. The examples are good places to see if we are doing it right. Basically we are re-exporting things at the crate root if they are "typical" Bitcoin terms, otherwise re-exporting from a well named submodule. Open to suggestions.